lib/varlink.c

```diff
@@ -396,6 +399,26 @@ pwaccess_check_expired(const char *user, long *daysleft, bool *pwchangeable, cha
   if (!user)
     return -EINVAL;
 
+  /*
+   * this comment is regarding all Varlink connections performed in the
+   * context of this PAM module.
+   * the Varlink service will use the UID returned by `sd_varlink_get_peer_uid(link, * &peer_uid);`
+   * for credentials checking.
+   * this UID will be the effective UID of the process seen at connect() time.
+   *
+   * If this PAM module runs e.g. in `sudo` context then the Varlink service
+   * will believe we are root and allow all operations.
```

Comment:
I double checked this again:
If `no_new_privs` is that there is never a problem.
Else:
In cases like sudo there is no problem, at least I couldn't find a situation where it makes a difference.

But there is one case were this can make a problem: if somebody sets the setuid bit for the new passwd, chfn or chsh commands, or in general, if somebody calls the chfn, chsh or chauthtok method from a setuid binary.
Easiest solution is to add code to this applications, that they refuse to work if a setuid bit is set.
If somebody explicit develops a setuid binary calling the three methods: I haven't found anything which could prevent this.